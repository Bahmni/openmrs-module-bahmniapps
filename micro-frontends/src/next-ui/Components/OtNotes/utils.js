import axios from 'axios';
import { OT_NOTES_BASE_URL } from "../../constants";

export const deleteOtNote = async (id) => {
    try {
        const response = await axios.delete(OT_NOTES_BASE_URL + "/" + id, {
            headers: {"Accept": "application/json", "Content-Type": "application/json", withCredentials: true}
        });
        return response.data;

    } catch (e) {
        console.error(e);
    }
}

export const constructPayload = function (noteDate, note, noteEndDate) {
    const payload = [];
    const currentDate = new Date(noteDate);
    while (currentDate <= noteEndDate) {
        payload.push({
            noteTypeName: "OT module",
            noteDate: new Date(currentDate),
            noteText: note
        });
        currentDate.setDate(currentDate.getDate() + 1);
    }
    return payload;
};

export const saveNote = async (note, startDate, endDate) => {
    try{
        const payload = endDate != null ? constructPayload(startDate, note, endDate) : [{
            noteTypeName: "OT module",
            noteDate: startDate,
            noteText: note
        }];
        const response = await axios.post(OT_NOTES_BASE_URL, payload);
        return response.data;
    }
    catch (e) {
        console.error(e);
    }
}

export const updateNoteForADay = async (noteId, note, providerUuid) => {
    const payload = {
        noteText: note,
        providerUuid: providerUuid
    };
    try{
        const response = await axios.post(OT_NOTES_BASE_URL + "/" + noteId, payload, {
            headers: {"Accept": "application/json", "Content-Type": "application/json"}
        });
        return response.data;
    }
    catch (e) {
        console.error(e);
    }
};