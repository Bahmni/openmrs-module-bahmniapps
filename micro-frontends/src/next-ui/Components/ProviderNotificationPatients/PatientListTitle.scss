.patient-list-tile-content {
    display: flex;
    align-items: center;
    padding: 10px;

    .patient-info {
        display: flex;
        flex-flow: row;
        justify-content: start;
        align-items: center;
        margin-left: 15px;
        font-size: .875rem;
    }

    a.bx--link.patient-id {
        color: #2b95ff;
        font-size: .875rem;
        margin-right: 10px;
    }
}

.warning {
    display: flex;
    color: red;
    align-items: center;
}


.drug-count {
    padding-left: 5px;
}