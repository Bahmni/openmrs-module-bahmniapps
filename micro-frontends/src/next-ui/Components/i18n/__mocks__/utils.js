export function getTranslations() {
  return import("../../../../../public/i18n/locale_en.json").then(
    (module) => module.default
  );
}

export const getLocale = () => "en";
