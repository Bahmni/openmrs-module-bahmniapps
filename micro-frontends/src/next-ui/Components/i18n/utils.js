import { LS_LANG_KEY, BASE_URL } from "../../constants";

const translationsBaseUrl = "i18n";

export function getLocale() {
  return localStorage.getItem(LS_LANG_KEY) || "en";;
}

export const getTranslations = async (locale) => {
  const fileName = `locale_${locale}.json`;
  return fetchTranslations(fileName);
};

async function fetchTranslations(fileName) {
  const url = `${BASE_URL}${translationsBaseUrl}/${fileName}`;
  const response = await fetch(url);
  return response.json();
}
