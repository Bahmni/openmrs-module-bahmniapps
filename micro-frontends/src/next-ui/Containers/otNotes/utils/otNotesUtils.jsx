import axios from 'axios';
import { OT_NOTES_BASE_URL } from "../../../constants";

export const deleteOtNote = async (id) => {
    try {
        const response = await axios.delete(OT_NOTES_BASE_URL + "/" + id, {
            headers: {"Accept": "application/json", "Content-Type": "application/json", withCredentials: true}
        });
        return response.data;

    } catch (e) {
        console.error(e);
    }
}