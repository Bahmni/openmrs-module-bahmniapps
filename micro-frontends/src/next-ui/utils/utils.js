import moment from "moment";
import { defaultDateTimeFormat } from "../constants";

export const formatDate = (value, format = defaultDateTimeFormat) => {
  return value ? moment(value).format(format) : value;
};
