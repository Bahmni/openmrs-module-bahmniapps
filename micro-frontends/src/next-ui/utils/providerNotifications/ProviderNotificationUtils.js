import axios from "axios";
import { EMERGENCY_MEDICATIONS_BASE_URL, GET_DRUG_ACKNOWLEDGEMENT_URL, GET_PROVIDER_UUID_URL } from "../../constants";
import { parseDateArray } from "../utils";

export const getPatientIPDDashboardUrl = (patientUuid, visitUuid) =>
    `/bahmni/clinical/index.html#/default/patient/${patientUuid}/dashboard/visit/ipd/${visitUuid}?source=clinical`;

export const getEmergencyDrugAcknowledgements = async (locationUuid, property, providerUuid) => {
    const apiURL = GET_DRUG_ACKNOWLEDGEMENT_URL.replace("{location_uuid}", locationUuid).replace("{property}", property).replace("{provider_uuid}",providerUuid);

    try {
        const response = await axios.get(apiURL);
        if (response.status === 200) {
            return response.data;
        }
        return [];
    } catch (error) {
        console.error(error);
    }
}

export const sortMedicationList = (medicationList) => {
    medicationList.forEach(group => {
        group.sort((a, b) => parseDateArray(a.administered_date_time).diff(parseDateArray(b.administered_date_time)));
    });
    medicationList.sort((a, b) => parseDateArray(a[0].administered_date_time).diff(parseDateArray(b[0].administered_date_time)));
    return medicationList;
};

export const acknowledgeEmergencyMedication = async (emergencyMedication, medicationAdministrationUuid) => {
    try {
        const response = await axios.put(
            EMERGENCY_MEDICATIONS_BASE_URL.replace("{medication_administration_uuid}",medicationAdministrationUuid),
            emergencyMedication
        );
        if (response.status === 200) {
            return { success: true, message: 'Emergency Medication Acknowledged Successfully' };
        } else {
            const errorMessage = response.data.error.message;
            return { success: false, message: `Error: ${errorMessage}` };
        }
    } catch (error) {
        return { success: false, message: `Error: ${error}` };
    }
};

export const getProvider = async () => {
    try {
        const response = await axios.get(GET_PROVIDER_UUID_URL);
        if (response.status === 200) {
            return response.data;
        }
        return [];
    } catch (error) {
        console.error(error);
    }
}

export const groupByIdentifier = (data, key) => {
    return data.reduce((accumulator, item) => {
      if (!accumulator[item[key]]) {
        accumulator[item[key]] = [];
      }
      accumulator[item[key]].push(item);
      return accumulator;
    }, {});
  };
  