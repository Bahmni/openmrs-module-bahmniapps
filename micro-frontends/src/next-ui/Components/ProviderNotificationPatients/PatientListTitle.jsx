import React from "react";
import { WarningAlt16 } from "@carbon/icons-react";
import { Link } from "carbon-components-react";
import "./PatientListTitle.scss";
import { getPatientIPDDashboardUrl } from "../../utils/providerNotifications/ProviderNotificationUtils";
import { formatGender } from "../../utils/utils";
import PropTypes from "prop-types";

const PatientListTitle = ({ noOfDrugs, identifier, name, age, gender, patientUuid, visitUuid }) => (
  <div className="patient-list-tile-content">
    <div className="warning">
      <WarningAlt16 />
      <span className="drug-count">{noOfDrugs}</span>
    </div>
    <div className="patient-info">
      <Link href={getPatientIPDDashboardUrl(patientUuid, visitUuid)} className="patient-id">
        {`(${identifier})`}
      </Link>
      <span>{`${name} - ${formatGender(gender)}, ${age}`}</span>
    </div>
  </div>
);

PatientListTitle.propTypes = {
  noOfDrugs: PropTypes.number.isRequired,
  identifier: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  age: PropTypes.string.isRequired,
  gender: PropTypes.string.isRequired,
  patientUuid: PropTypes.string.isRequired,
  visitUuid: PropTypes.string.isRequired,
};

export default PatientListTitle;
