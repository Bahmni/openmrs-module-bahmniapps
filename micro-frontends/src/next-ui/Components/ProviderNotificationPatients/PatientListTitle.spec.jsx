import React from "react";
import { render } from "@testing-library/react";
import PatientListTitle from "./PatientListTitle";
import { formatGender } from "../../utils/utils";
import { getPatientIPDDashboardUrl } from "../../utils/providerNotifications/ProviderNotificationUtils";

const mockWindowOpen = jest.fn();
global.window.open = mockWindowOpen;

jest.mock("../../utils/providerNotifications/ProviderNotificationUtils", () => ({
  getPatientIPDDashboardUrl: jest.fn(),
}));

describe("PatientListTitle Component", () => {

  beforeEach(()=>{
    jest.clearAllMocks();
  })

  const props = {
    noOfDrugs: 3,
    identifier: "PID123",
    name: "John Doe",
    age: "30",
    gender: "M",
    patientUuid: "patient123",
    visitUuid: "visit123",
    openedWindow: null,
    setOpenedWindow: jest.fn(),
  };

  it('should render without crashing', () => {
    const {container} = render(<PatientListTitle {...props} />);
    expect(container).toMatchSnapshot();
  })

  it('renders the correct number of drugs', () => {
    const { getByText } = render(<PatientListTitle {...props} />);
    expect(getByText(`${props.noOfDrugs}`)).toBeTruthy();
  });

  it('renders the patient identifier within a Link', () => {
    const expectedUrl = 'http://example.com/dashboard';
    getPatientIPDDashboardUrl.mockReturnValue(expectedUrl);
    const { getByText } = render(<PatientListTitle {...props} />);
    const link = getByText(`(${props.identifier})`);
    expect(link.href).toBe(expectedUrl);
    expect(link.textContent).toBe(`(${props.identifier})`);
  });

  it('renders the patient name, gender, and age', () => {
    const { getByText } = render(<PatientListTitle {...props} />);
    const expectedText = `${props.name} - ${formatGender(props.gender)}, ${props.age}`;
    expect(getByText(expectedText)).toBeTruthy();
  });

  it('passes prop types correctly', () => {
    const consoleError = jest.spyOn(console, 'error').mockImplementation(() => {});
    const invalidProps = {
      noOfDrugs: '3',
    };
    render(<PatientListTitle {...invalidProps} />);
    expect(consoleError).toHaveBeenCalled();
    consoleError.mockRestore();
  });
});
