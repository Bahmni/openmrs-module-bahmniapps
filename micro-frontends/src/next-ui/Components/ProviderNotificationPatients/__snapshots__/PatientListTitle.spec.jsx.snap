// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PatientListTitle Component should render without crashing 1`] = `
<div>
  <div
    class="patient-list-tile-content"
  >
    <div
      class="warning"
    >
      <svg
        aria-hidden="true"
        fill="currentColor"
        focusable="false"
        height="16"
        preserveAspectRatio="xMidYMid meet"
        viewBox="0 0 16 16"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M14.5,14h-13c-0.2,0-0.3-0.1-0.4-0.2c-0.1-0.2-0.1-0.3,0-0.5l6.5-12C7.7,1,8,0.9,8.2,1.1c0.1,0,0.2,0.1,0.2,0.2l6.5,12	c0.1,0.2,0.1,0.3,0,0.5C14.9,13.9,14.7,14,14.5,14z M2.3,13h11.3L8,2.5L2.3,13z"
        />
        <path
          d="M7.5 6H8.5V9.5H7.5zM8 10.8c-.4 0-.8.3-.8.8s.3.8.8.8c.4 0 .8-.3.8-.8S8.4 10.8 8 10.8z"
        />
      </svg>
      <span
        class="drug-count"
      >
        3
      </span>
    </div>
    <div
      class="patient-info"
    >
      <a
        class="bx--link patient-id"
      >
        (PID123)
      </a>
      <span>
        John Doe - Male, 30
      </span>
    </div>
  </div>
</div>
`;
